### DISCUSSION TITLE
Starfield speed

### DISCUSSION THREAD

**Discussion by daxliniere** [Bodmer/TFT_eSPI#1425] [General]
Hey everyone,
Any suggestions on how to slow down the starfield sketch? I added a delay, but of course that just made the motion chunky.

(Also, I think I have to apologise for the method I used to propose a change via the Pull Request system. I hope that doesn't cause too much grief for you @Bodmer!)

All the best,
Dax.

**Comment by Bodmer** [comment]
There are a few ways to do this:
1. Increase the number of stars (i.e. NSTARS) but be aware of the memory demands since each start uses 3 bytes of RAM
2. Change the rate at which the depth coordinate z (i.e aparrent distance) of each star changes, so [change this line](https://github.com/Bodmer/TFT_eSPI/blob/master/examples/320%20x%20240/TFT_Starfield/TFT_Starfield.ino#L65) to:
```
      sz[i] -= 1;
```
  That will halve the "speed" of the star movement and make pixel move distances smaller, hence reducing movement jerkiness
3. Add a delay after [drawing each pixel](https://github.com/Bodmer/TFT_eSPI/blob/master/examples/320%20x%20240/TFT_Starfield/TFT_Starfield.ino#L75) (i.e star) by adding a small delay e.g.:
```
          tft.drawPixel(screen_x, screen_y, tft.color565(r,g,b));
          delayMicroseconds(50);
```

**Reply by daxliniere** [reply]
Thanks @Bodmer. Solution #2 seems like the best one. I tried some values <1, but it made no difference. Then I realised that this would require some kind of sub-pixel sampling which seems outside the scope of this simple demo.

I only put time into this because of my fondness for the old ssstarfield.scr which I still use to this day. Call me ol' skool. 😉 
Cheers!
Dax.
